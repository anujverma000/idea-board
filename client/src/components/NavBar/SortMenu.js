import React, { useState, useRef, useEffect } from "react";
import SortOption from './SortOption'
import styled from "styled-components";

const SortMenu = ({ sort }) => {
  const [ displayMenu ,setDisplayMenu ] = useState(false);
  const node = useRef();

  useEffect(() => {
    document.addEventListener('mouseup', handleClickOutside)
    return () => {
      document.removeEventListener('mouseup', handleClickOutside)
    }
  },[])
  
  const handleClickOutside = e => {
    if (node.current && !node.current.contains(e.target)) {
      setDisplayMenu(false)
    }
  }

  return(
    <Wrapper ref={ node } onClick={() => setDisplayMenu(!displayMenu) }>
      <Sort>Sort</Sort>
      { displayMenu &&
        <OptionsMenu>
          <SortOption field='title' sort={sort} label='Title'/>
          <SortOption field='date' sort={sort} label='Date'/>
        </OptionsMenu>
      }
    </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.div`
  position: relative;
`;
const Sort = styled.div`
  color: ${props => props.theme.textColorPrimary};
  padding: 16px 24px;
  font-size: 16px;
  border-radius: 4px;
  cursor: pointer;
  margin: 0 16px;
  min-width: 150px;
  box-sizing: border-box;
  text-align: center;
  border: 1px solid ${props => props.theme.textColorPrimary};
  background-color: ${props => props.theme.backgroundColor};
  &:hover{
      transition: .3s;
      color: ${props => props.theme.textColorInvert};
      background-color: ${props => props.theme.primaryColorLight};
  }
  &:active{
      color: ${props => props.theme.primaryColor};
  }
`;

const OptionsMenu = styled.div`
  background-color: ${props => props.theme.backgroundColor};
  display: flex;
  position: absolute;
  top: 60px;
  flex-direction: column;
  min-width: 200px;
  border: 1px solid ${props => props.theme.borderColor};
  border-radius: 4px;
  z-index: 3;
  box-shadow: 0 0 12px 4px ${props => props.theme.shadowColor};
`;

export default SortMenu;