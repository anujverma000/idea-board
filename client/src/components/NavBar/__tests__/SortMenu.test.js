import React from 'react';
import renderer from 'react-test-renderer';
import SortMenu from '../SortMenu'

it('renders correctly', () => {
  const tree = renderer.create(<SortMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
