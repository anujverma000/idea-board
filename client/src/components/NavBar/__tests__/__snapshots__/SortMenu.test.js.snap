// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-ifAKCX bqyzQp"
  onClick={[Function]}
>
  <div
    className="sc-EHOje dFQdKu"
  >
    Sort
  </div>
</div>
`;
